codes/typescript/chapter_searching/hashing_search.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * File: hashing_search.js
+ * Created Time: 2022-12-29
+ * Author: zhuoqinyue (1403450829@qq.com)
+ */
+
+/* 哈希查找（数组） */
+function hashingSearch(map: Map<number, number>, target: number) {
+  // 哈希表的 key: 目标元素，value: 索引
+  // 若哈希表中无此 key ，返回 -1
+  return map.has(target) ? map.get(target) : -1;
+}
+
+/* 哈希查找（链表） */
+function hashingSearch1(map: Map<number, any>, target: number) {
+  // 哈希表的 key: 目标结点值，value: 结点对象
+  // 若哈希表中无此 key ，返回 null
+  return map.has(target) ? map.get(target) : null;
+}
+
+function main() {
+  const target = 3;
+
+  /* 哈希查找（数组） */
+  const nums = [1, 5, 3, 2, 4, 7, 5, 9, 10, 8];
+  // 初始化哈希表
+  const map = new Map();
+  for (let i = 0; i < nums.length; i++) {
+    map.set(nums[i], i);  // key: 元素，value: 索引
+  }
+  const index = hashingSearch(map, target);
+  console.log("目标元素 3 的索引 = " + index);
+
+//   /* 哈希查找（链表） */
+//   let head = new ListNode().arrToLinkedList(nums)
```

Comment:
Uncomment them

Reply:
Maybe i will delete them.
There no exists the arrToLinkedList method in `typescript\module\ListNode.ts`.
I could translate this method in next week.

Reply:
OK. Feel free to add codes to this PR.